WatermarkDetectron.py

```diff
@@ -23,35 +29,50 @@ def __load_model(self, pretrained=None):
 
         # TODO
 
-        return ...
+        return None
     
     """
     Load an image from an URL address
     @param url: URL address of the image
     @return A generator of this image 
     """
     def __read_file_from_url(self, url):
-
-        # TODO
-        
-        return ...
+        if url.endswith('.png'):
+            img_tensor = tf.image.decode_png(requests.get(url).content, channels=3, name="png_reader")
+            img_tensor = tf.image.resize(img_tensor, [200, 200])
+        elif url.endswith('.jpg') or url.endswith('.jpeg'):
+            img_tensor = tf.image.decode_jpeg(requests.get(url).content, channels=3, name="jpeg_reader")
+            img_tensor = tf.image.resize(img_tensor, [200, 200])
+        else:
```

Comment:
I'm thinking if we can also include a case for gif images. Is it possible for them to have watermarks at the bottom-right corner for gifs?

Reply:
A possible solution is to keep the first frame of the gif as the reference, then process that just as a static image. But need some researches on this...